drivers/adc/Kconfig.gecko

```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2023 Antmicro <www.antmicro.com>
 # SPDX-License-Identifier: Apache-2.0
 
-config ADC_GECKO
+config IADC_GECKO
```

Comment:
I'd suggest to use `ADC_GECKO_IADC` for consistency with other ADC related Kconfig options (see for example `ADC_MCUX_*` or `ADC_NRFX_*`).

Reply:
Config has been renamed per your suggestion